### DISCUSSION TITLE
Trying to get a boolean based on if the "device" is on or off.

### DISCUSSION THREAD

**Discussion by kennydossantos** [HomeSpan/HomeSpan#664] [Q&A]
Hi, I can't figure out how to use HomeSpan to create for example a virtual lamp to save the sate of the lamp (false or true) to a variable, so I can later use that boolean in my code. For example opening a door using another function. So I would like HomeSpan to save something and if it is true than open door. I have tried myself (see below), but I can't get it to work. I would appreciate all the help I could get.

`#include "HomeSpan.h"

HomeSpan homeKitAccessory("MyAccessory", "AccessoryType");

bool powerState = false;  
Characteristic::On powerCharacteristic("Power", powerState);

void handlePowerChange(bool newValue) {
    if (newValue) {
        MyFunction();
    }
}

void setup() {
    homeKitAccessory.addCharacteristic(&powerCharacteristic);

    powerCharacteristic.onValueChange(std::bind(handlePowerChange, std::placeholders::_1));

    homeKitAccessory.begin();
}

void loop() {
    homeKitAccessory.loop();
}
`

**Comment by frankonski** [comment]
The methods you are calling are not part of this project here, HomeSpan. `addCharacteristic()` seems to belong to a completely different project.
